README.md
# Examen 372 - Terminal / Git-Github 👻💻

 > " Solo hay un riesgo que deberías evitar a toda costa, y es el riesgo de no hacer nada 💪🔥" *(Denis Waitley)*


### Instrucciones Generales 🤔: 

1. Todas las instrucciones deben ser ejecutadas en terminal a menos que se indique lo contrario.
2. Se permite el uso de internet alias el **"googlazo"**.
3. La entrega del examen es un repositorio en su cuenta de GITHUB con la solución.
4. Cada inciso del examen debe ser marcado como completado para verificar que se ha realizado correctamente.
5. Enviar correo a iavila24@uabc.edu.mx con el link del repositorio para verificar examen.
6. Límite de entrega **`24-OCT-2023 20:59:59`**
---

### Intrucciones Examen 🧐:

- [X] Coloca tu nombre aqui: *`"Jennifer Lizeth Damaso Negrete"`*
- [X] Abre el shell/terminal *`"Git Bash"`*
- [X] Limpia el historial de tu consola: *`"history -c"`*
- [X] Navega al directorio home del usuario. *`"~"`* 
- [X] Crear una carpeta con el nombre. *`"examen372"`* y colocate dentro de ella.
- [X] Crear un script con el nombre. *`"create.sh"`*
- [X] Colocar el shebang dentro del script desde la terminal: *`"#! /bin/bash"`*
- [X] Inicializa un repositorio vacio de git.
- [X] Crea un commit inicial con el mensaje *`"Initial commit, script added"`* (el commit debe contener la version inicial del script).
---

### Instrucciones dentro del script `create.sh` 🔡:

1. [X] Elimina la carpeta **webapp**. *`"rm -rf"`*
2. [X] Crea la siguiente estructura de carpetas *`"mkdir"`*:
  
    ```
    .
    └── webapp/
        ├── css
        ├── js
        ├── docs/
        │   ├── text
        │   └── python
        └── img
    ```
    
3. [X] Crea los archivos de la aplicación con el siguiente código (incluyelo en el script):
    
    ```sh
    for file in index.html style.css app.js main.py
    do
        touch ./"${file}"
    done
    ```

4. [X] Una vez creados los archivos la carpeta se vera asi utiliza el siguiente comando si quieres ver en consola tu estructura `cmd //c tree //F ./`:

    ```
    .
    └── webapp/
        ├── app.js
        ├── index.html
        ├── main.py
        ├── style.css
        ├── css
        ├── docs/
        │   ├── text
        │   └── python
        ├── img
        └── js

    ```

5. [X] Descargar la imagen de internet con el siguiente comando:
    
    ```sh
    curl -o meme.jpg "https://pm1.aminoapps.com/7021/417429318f6b29b6bc737fb95ba1ddafc9ba2decr1-720-642v2_hq.jpg"
    ```
6. [X] Guarda en una variable la cantidad de archivos de texto que creara el script y pon el siguiente código esto creara la cantidad de archivos .txt ingresada por consola:
    
    ```sh
    read -p "Introduce la cantidad de archivos a crear:" numfiles
    for x in $( seq 1 $numfiles )
    do
        echo "Archivo ${x}" > "file${x}.txt"
    done
    ```

7. [X] Verificar estructura con `"cmd //c tree //F ./"` debe verse de la siguiente manera

    ```
    .
    └── webapp/
        ├── app.js
        ├── file1.txt
        ├── file2.txt
        ├── file3.txt
        ├── file4.txt
        ├── file5.txt
        ├── ...
        ├── ...
        ├── fileN.txt
        ├── index.html
        ├── main.py
        ├── meme.jpg
        ├── style.css  
        ├── css
        ├── docs/
        │   ├── text
        │   └── python
        ├── img
        └── js

    ```
   
8. [X] En este punto crea un segundo commit con el mensaje *`"Second commit, structure created"`* que contiene la actualización del script (este comando no debe ser parte del script).
   
9. [X] En el script debe organizar la informacion moviendo los archivos a su respectiva carpeta usando el comando *`"mv"`*:
   
      - [X] Mueve el archivo *`style.css`* a la carpeta **`css`**
      - [X] Los archivos *`fileN.txt`* a la carpeta **`docs/text`**
      - [X] El archivo *`main.py`* a la carpeta **`docs/python`**
      - [X] El *`app.js`* a la carpeta **`js`** 
      - [X] Dejar *`index.html`* en la ruta actual, dentro de **`webapp`**. 
   
10. [X] Al ejecutar `"cmd //c tree //F ./"` debe verse asi y concluye la programación del script:
   
    ```
    .
    └── webapp/
        ├── index.html
        ├── css/
        │   └── style.css
        ├── docs/
        │   ├── text/
        │   │   ├── file1.txt
        │   │   ├── file2.txt
        │   │   ├── file3.txt
        │   │   ├── file4.txt
        │   │   ├── file5.txt
        │   │   ├── ...
        │   │   ├── ...
        │   │   └── fileN.txt
        │   └── python/
        │       └── main.py
        ├── img/
        │   └── meme.jpg
        └── js/
            └── app.js
        
    ```
---
- [X] Una vez completado el script ejecutarlo para validar que haga lo antes pedido: `./create.sh`
- [X] Mover este archivo `examen372.md` a la carpeta **`examen372`**
- [X] Renombrar el archivo `examen372.md` a README.md con la terminal *`"mv"`*
- [X] Registrar un tercer commit con el mensaje *`"Third commit, add README.md and final script updated"`*
- [X] Guardar el historial de sus comandos: *`history > history.txt`*
- [X] Registrar el último commit agregando el archivo del historial con el mensaje *`"Fourth commit, command history added"`*
- [X] Crear un repositorio remoto en sus cuentas de GITHUB con el nombre *`examen372`*
- [X] Subir su carpeta al repositorio remoto con su famosisimo: *`git push`* 🔥👀

---

### Extras:

<span style="font-size:1.5em">**+1 extra**</span> en la calificación del examen si colocas el código siguiente en los archivos de la app.

> index.html

```html
<!DOCTYPE html>
<html lang="es">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Cheems I love you 💖🐶</title>
    <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Lato:wght@400;700&display=swap">
    <link rel="stylesheet" href="./css/styles.css">
    
</head>
<body>
    <main id="content">
        <img id="cheems-img" src="https://img.asmedia.epimg.net/resizer/zCBy1qXw2E26HUAAQTBOYB0eSko=/736x414/cloudfront-eu-central-1.images.arcpublishing.com/diarioas/KKF52TVGVNCKJFW6RGQRBR6OBE.png" alt="Perrito Cheems">
    </main>
    
    <script src="https://cdn.jsdelivr.net/npm/tsparticles-confetti@2.12.0/tsparticles.confetti.bundle.min.js"></script>
    <script src="./js/app.js"></script>
</body>
</html>
```

> style.css

```css
body {
    font-family: 'Roboto', sans-serif;
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100vh;
    margin: 0;
    background-color: #333;
}

#cheems-img {
    cursor: pointer;
}

h1 {
    font-size: 4rem; /* Tamaño de fuente de 4rem */
    color: #fff; /* Texto blanco */
    text-shadow: 2px 2px 4px rgba(0, 0, 0, 0.5); /* Sombra de texto */
}
```

> app.js
> 
```js
// Add envent click to image
const content = document.getElementById("content");
const cheemsImg = document.getElementById("cheems-img");
const h1 = document.createElement("h1");
h1.textContent = "¡Felicidades Cheems 😭!";

cheemsImg.addEventListener("click", function() {
    cheemsImg.style.display = "none"; // Hide image
    content.appendChild(h1); // Add h1 element
    explode();
    
});

// Function to make explode confetti
function explode() {
    const duration = 15 * 1000,
    animationEnd = Date.now() + duration,
    defaults = { startVelocity: 30, spread: 360, ticks: 60, zIndex: 0 };

    function randomInRange(min, max) {
    return Math.random() * (max - min) + min;
    }

    const interval = setInterval(function() {
    const timeLeft = animationEnd - Date.now();

    if (timeLeft <= 0) {
        return clearInterval(interval);
    }

    const particleCount = 50 * (timeLeft / duration);

    // since particles fall down, start a bit higher than random
    confetti(
        Object.assign({}, defaults, {
        particleCount,
        origin: { x: randomInRange(0.1, 0.3), y: Math.random() - 0.2 },
        })
    );
    confetti(
        Object.assign({}, defaults, {
        particleCount,
        origin: { x: randomInRange(0.7, 0.9), y: Math.random() - 0.2 },
        })
    );
    }, 250);
}
```

`Nota:` Si descargo tu script y ejecuta la app tienes <span style="font-size:1.5em">**+2 extra**</span>.
<br />
<br />
<br />
<div align="center" width="250" height="250">

![Examen](https://www.elorientadero.com/wp-content/uploads/2022/01/meme-1.jpg)

</div>

history.txt
    1  history
    2  cd ..
    3  mkdir examen372
    4  cd examen372
    5  touch create.sh
    6  echo "#! /bin/bash" > create.sh
    7  git init
    8  git status
    9  git add create.sh
   10  git commit -m "Initial commit, script added"
   11  code create.sh
   12  ./create.sh
   13  cmd //c tree //F ./
   14  ./create.sh
   15  cmd //c tree //F ./
   16  ./create.sh
   17  cmd //c tree //F ./
   18  git status
   19  git add create.sh
   20  git commit -m "Second commit, structure created"
   21  git status
   22  ./create.sh
   23  cmd //c tree //F ./
   24  ./create.sh
   25  cmd //c tree //F ./
   26  ./create.sh
   27  cmd //c tree //F ./
   28  mv examen372.md ./examen372
   29  cd ..
   30  mv examen372.md ./examen372
   31  cd Examen1
   32  ls
   33  mv examen372.md ./examen372
   34  ls
   35  mv ../ examen372
   36  mv examen372 ../
   37  ls
   38  mv ./examen372 ../
   39  mv ./examen372 ~/
   40  ls~
   41  ls ~
   42  mv examen372.md ./examen372
   43  mv examen372.md /c/Users/Lizeth Damaso/examen372
   44  mv examen372.md /c/Users/Lizeth Damaso/examen372/
   45  mv examen372.md 
   46  mv examen372.md 
   47  mv examen372.md /C/users/Lizeth\ Damaso/examen372
   48  mv examen372.md /C/users/Lizeth\ Damaso/examen372/
   49  mv examen372.md /C/users/
   50  mv examen372.md /c/users/
   51  mv examen372.md /C/users/Lizeth\ Damaso/examen372/
   52  mv examen372.md README.md
   53  cd ..
   54  cd examen372
   55  mv examen372.md README.md
   56  git status
   57  git add create.sh
   58  git commit -m "Third commit, add README.md and final script updated"
   59  git add README.md
   60  git commit -m "Third commit, add README.md and final script updated"
   61  history > history.txt


create.sh
#! /bin/bash

rm -rf webapp

# ESTRUCTURA DE CARPETAS
mkdir webapp
cd webapp
mkdir css
mkdir js
mkdir docs
cd docs
mkdir text
mkdir python
cd ..
mkdir img

# Crear archivos de la aplicacion
for file in index.html style.css app.js main.py
    do
        touch ./"${file}"
    done

# Estructura de las carpetas en la terminal
   # cmd //c tree //F ./

# Descargar imagen
   curl -o meme.jpg "https://pm1.aminoapps.com/7021/417429318f6b29b6bc737fb95ba1ddafc9ba2decr1-720-642v2_hq.jpg"

   #variable 
read -p "Introduce la cantidad de archivos a crear:" numfiles
    for x in $( seq 1 $numfiles )
    do
        echo "Archivo ${x}" > "file${x}.txt"
    done

    #Mover archivos a sus carpetas
    mv style.css ./css
    mv file*.txt ./docs/text
    mv main.py ./docs/python
    mv app.js ./js
    mv meme.jpg ./img